internal/pkg/repository/graph/elastic_graph_impl.go

```diff
@@ -0,0 +1,244 @@
+}
+func (p *GraphElasticRepository) FetchElasticFarmPoolsFromGraphByIDs(addresses []string) ([]entity.ElasticFarmPool, error) {
+	ids := make([]string, 0, len(addresses))
+	for i, _ := range addresses {
+		ids = append(ids, "'"+addresses[i]+"'")
+	}
+	var farmPools []entity.ElasticFarmPool
+	query := fmt.Sprintf(`select f.pool, max(f.vid) as vid, max(f.id) as id, max(f.block) as block, max(f.farm) as farm, max(f.pid) as pid, max(f.start_time) as start_time, max(f.end_time) as end_time, max(f.fee_target) as fee_target, max(f.vesting_duration) as vesting_duration, ARRAY_AGG(f.token) as reward_tokens, ARRAY_AGG(f.amount) as total_reward_amounts from 
+		(SELECT grouped.vid, grouped.id, lower(t.block_range) as block
+		, t.farm, t.pid, t.start_time, t.end_time, t.fee_target, t.vesting_duration, t.pool
+		, rt.token, rt.amount
+		FROM 
+		(
+			SELECT id, max(vid) as vid FROM sgd1.farming_pool
+			GROUP BY id 
+			ORDER BY vid
+		) AS grouped
+		JOIN sgd1.farming_pool AS t ON grouped.vid=t.vid
+		JOIN sgd1.reward_token AS rt ON rt.farming_pool = t.id 
+		WHERE t.id in (%s)
+		ORDER BY block DESC) as f
+		group by f.pool`, strings.Join(ids, ", "))
+	res := p.db.Raw(query).Scan(&farmPools)
+	// res := p.GroupedThenRaw(p.getTable("farming_pool")).Scan(&farmPools)
+	return farmPools, res.Error
+}
+
+func (p *GraphElasticRepository) FetchElasticJoinedPositionsFromGraph() ([]entity.ElasticJoinedPosition, error) {
+	var joinedPositions []entity.ElasticJoinedPosition
+	res := p.GroupedThenRaw(p.getTable("joined_position")).Scan(&joinedPositions)
+	return joinedPositions, res.Error
+}
+
+func (p *GraphElasticRepository) FetchElasticJoinedPositionsFromGraphByIDs(ids []string) ([]entity.ElasticJoinedPosition, error) {
+	var joinedPositions []entity.ElasticJoinedPosition
+	res := p.GroupedThenRawByAddresses(p.getTable("joined_position"), ids).Scan(&joinedPositions)
+	return joinedPositions, res.Error
+}
+
+func (p *GraphElasticRepository) FetchElasticDepositedPositionsFromGraph() ([]entity.ElasticDepositedPosition, error) {
+	var depositedPositions []entity.ElasticDepositedPosition
+	res := p.GroupedThenRaw(p.getTable("deposited_position")).Scan(&depositedPositions)
+	return depositedPositions, res.Error
+}
+
+func (p *GraphElasticRepository) FetchElasticDepositedPositionsFromGraphByIDs(ids []string) ([]entity.ElasticDepositedPosition, error) {
+	var depositedPositions []entity.ElasticDepositedPosition
+	res := p.GroupedThenRawByAddresses(p.getTable("deposited_position"), ids).Scan(&depositedPositions)
+	return depositedPositions, res.Error
+}
+
+func (p *GraphElasticRepository) FetchElasticEventsFromBlock(blockNumber string) ([]entity.Event, error) {
+	var events []entity.Event
+	res := p.db.Raw(fmt.Sprintf(`
+	SELECT grouped.vid, grouped.id, lower(t.block_range) as block, t.*
+	FROM 
+	(
+		SELECT id, max(vid) as vid from %s 
+		GROUP BY id 
+		ORDER BY vid
+	) as grouped
+	JOIN %s AS t ON grouped.vid=t.vid
+	WHERE lower(t.block_range) > %s 
```

Comment:
be careful as comparing string number! eg: "25" is greater than "100"!

Reply:
I think it's int comparing, isn't it? 
```suggestion
	SELECT grouped.vid, grouped.id, lower(t.block_range) as block, t.*
FROM 
(
        SELECT id, max(vid) as vid from sgd1.contract_event 
        GROUP BY id 
        ORDER BY vid
) as grouped
JOIN sgd1.contract_event AS t ON grouped.vid=t.vid
WHERE lower(t.block_range) > 15716426 
ORDER BY block DESC
```

Reply:
I think it's int comparing, isn't it 
`SELECT grouped.vid, grouped.id, lower(t.block_range) as block, t.*
        FROM 
        (
                SELECT id, max(vid) as vid from sgd1.contract_event 
                GROUP BY id 
                ORDER BY vid
        ) as grouped
        JOIN sgd1.contract_event AS t ON grouped.vid=t.vid
        WHERE lower(t.block_range) > 15716426 
        ORDER BY block DESC`

Reply:
`lower()` is a method for string, it's a bit weird as seeing `WHERE lower(t.block_range) >`